### PR TITLE
improve cross shard tx receipts verification design and logic

### PR DESCRIPTION
**Pull request by chaosma**
- add header and signature information when broadcasting cxreceipts
- avoid crosslink verification to speedup cross shard tx finalization by around 1 block time.

### CODE REVIEW

**Comment by rlan35** [comment]
I think directly sending signature may not work in the case two shards change epoch at different time.

File: node/node_newblock.go [hunk 1]
```diff
@@ -218,18 +217,13 @@ func (node *Node) proposeReceiptsProof() []*types.CXReceiptsProof {
...
-		if err := node.compareCrosslinkWithReceipts(cxp); err != nil {
-			utils.Logger().Debug().Err(err).Interface("cxp", cxp).Msg("[proposeReceiptsProof] CrossLink Verify Fail")
-			if err != ErrCrosslinkVerificationFail {
-				pendingReceiptsList = append(pendingReceiptsList, cxp)
-			}
-		} else {
-			utils.Logger().Debug().Interface("cxp", cxp).Msg("[proposeReceiptsProof] CXReceipts Added")
-			validReceiptsList = append(validReceiptsList, cxp)
-		}
+		utils.Logger().Debug().Interface("cxp", cxp).Msg("[proposeReceiptsProof] CXReceipts Added")
```
Comment (rlan35) [inline node/node_newblock.go]: Why don't we do "isvalidCX..." check in proposal? I think it's better to check here.

File: node/node_newblock.go [hunk 2]
```diff
@@ -218,18 +217,13 @@ func (node *Node) proposeReceiptsProof() []*types.CXReceiptsProof {
...
+		validReceiptsList = append(validReceiptsList, cxp)
```
Comment (rlan35) [inline node/node_newblock.go]: We should have a upper limit on the number of receipts in one block.

Comment (rlan35) [reply node/node_newblock.go]: on the source: 2000 and here: 6000 limit.

File: node/node_newblock.go (hunk 1 above)
Comment (chaosma) [reply node/node_newblock.go]: move it in proposal

File: node/node_newblock.go (hunk 2 above)
Comment (chaosma) [reply node/node_newblock.go]: add upperlimit on incoming receipts as well as outgoing receipts

**Comment by chaosma** [comment]
Jenkins, shut up! Test this, please!

**Pull request merged**